### PR TITLE
Instance group ips build task

### PR DESCRIPTION
**Pull request by ILMTitan**

### CODE REVIEW

File: DETAILS.md [hunk 1]
```diff
@@ -40,6 +40,8 @@ The extension installs the following tasks:
   - **[Get Google Compute VM password][set-login-password]**: Gets the IP address
   and user password for a [Google Compute Engine][ComputeEngine] virtual machine
   running windows.
+  - **[Instance Group IPs][instance-group-ips]**: Collect the external IPs of an [instance group][InstanceGroup]
```
Comment (quoctruong) [inline DETAILS.md]: nit: Should be "Collects" and "saves" to be consistent with the other ones.

File: common/handle-rejection.ts [hunk 2]
```diff
@@ -44,11 +44,20 @@ export async function catchAll<T>(promise: PromiseLike<T>|
     let message: string;
     if (e instanceof AggregateError) {
       message = e.message;
+      for (const stackLine of e.stack.split(/\n/)) {
+        task.debug(stackLine);
```
Comment (quoctruong) [inline common/handle-rejection.ts]: Is there a way to factor out the splitting of line logic?

File: instance-group-ips-build-task/instance-group-ips.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+ * The implementation for finding an instance group and getting the external
+ * IPs of its member virtual machines.
+ */
+export async function getInstanceGroupIps(
+    {
+      endpoint,
+      locationScope,
+      location,
+      instanceGroupName,
+      separator = ',',
```
Comment (quoctruong) [inline instance-group-ips-build-task/instance-group-ips.ts]: nit: I think optional argument should be moved towards the end. So `buildVariableName` should be in front of `separator`.

File: instance-group-ips-build-task/instance-group-ips.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  di?: {task: typeof task};
+}
+
+/**
+ * The implementation for finding an instance group and getting the external
+ * IPs of its member virtual machines.
+ */
+export async function getInstanceGroupIps(
+    {
+      endpoint,
```
Comment (quoctruong) [inline instance-group-ips-build-task/instance-group-ips.ts:58]: nit: This parameter name is easily confused with `Endpoint`.

File: instance-group-ips-build-task/manifest.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      "path": "run.js"
+    },
+    {
+      "path": "task.json"
+    },
+    {
+      "path": "icon.png"
+    }
+  ]
+}
```
Comment (quoctruong) [inline instance-group-ips-build-task/manifest.json]: nit: new line.

File: DETAILS.md (hunk 1 above)
Comment (ILMTitan) [reply DETAILS.md]: Done.

File: instance-group-ips-build-task/instance-group-ips.ts (hunk 3 above)
Comment (ILMTitan) [reply instance-group-ips-build-task/instance-group-ips.ts]: Done.

File: instance-group-ips-build-task/instance-group-ips.ts (hunk 4 above)
Comment (ILMTitan) [reply instance-group-ips-build-task/instance-group-ips.ts:58]: It is an instance of `Endpoint`.

File: instance-group-ips-build-task/manifest.json (hunk 5 above)
Comment (ILMTitan) [reply instance-group-ips-build-task/manifest.json]: Done.

File: common/handle-rejection.ts (hunk 2 above)
Comment (ILMTitan) [reply common/handle-rejection.ts]: Do you want a separate variable that is the array of split lines, or are you concerned about the duplication of code in the `Error` and `AggregateError` blocks?

**Comment by ILMTitan** [comment]
PTAL.

**Review by quoctruong — APPROVED** [review]
LGTM. Just one comment about the inner error messages handling.

**Pull request merged**